apps/qubit/modules/object/actions/exportAction.class.php

```diff
@@ -23,33 +23,121 @@ class ObjectExportAction extends DefaultEditAction
+    $items = $this->context->user->getClipboard()->countByType();
+    if(empty($items) || !array_key_exists($className, $items) || 1 > $items[$className])
+    {
+      $this->redirect($this->context->routing->generate(null, array('module' => 'user', 'action' => 'clipboard')) . '?type=' . $this->objectType);
+    }
+    // Get format and validate
+    // Currently 'switch' to process the inbound parameter, validate and set default - could be if/then if preferred
+    $this->formatType = trim(strtolower($request->getParameter('format')));
+    if(!('xml' == $this->formatType || 'csv' == $this->formatType) || 'repository' == $this->objectType)
+    {
+        $this->formatType = 'csv';
+    }
+
+    // Basic permission check to determine whether digital object export should be made available
+    $this->digitalObjectsAvailable = false;
+    if (sfConfig::get('app_clipboard_export_digitalobjects_enabled', false) && ('informationObject' == $this->objectType || ('actor' == $this->objectType && $this->context->user->isAuthenticated())))
+    {
+      $this->digitalObjectsAvailable = true;
+    }
+
+    // Show export options panel if:
+    // information object type
+    // or, if actor type and digital objects are on the clipboard
+    $this->showOptions = 'informationObject' == $this->objectType || ('actor' == $this->objectType && $this->digitalObjectsAvailable);
+
+    // Get field includeDescendants if:
+    // options enabled
+    // and, information object type
+    $this->descendantsIncluded = $this->showOptions && 'informationObject' == $this->objectType && 'on' == $request->getParameter('includeDescendants');
+    // get field includeAllLevels if:
+    // descendantsIncluded enabled
+    $this->descendantsAllLevels = $this->descendantsIncluded && 'on' == $request->getParameter('includeAllLevels');
+
+    // Get field includeDigitalObjects if:
+    // digital objects are on the clipboard
+    // and, information object type and descendants are not selected
+    // or, if actor type
+    $this->includeDigitalObjects = false;
+    if ($this->digitalObjectsAvailable)
     {
-      $this->redirectUrl = $referrer;
+      $this->includeDigitalObjects = 'on' == $request->getParameter('includeDigitalObjects');
     }
 
+    // Get field includeDrafts if:
+    // options enabled
+    // and, user is authenticated
+    $this->draftsIncluded = false;
+    if ($this->showOptions && $this->context->user->isAuthenticated())
+    {
+      $this->draftsIncluded = 'on' == $request->getParameter('includeDrafts');
+    }
+
+    parent::execute($request);
+
+    $this->response->addJavaScript('exportOptions', 'last');
+
     $this->title = $this->context->i18n->__('Export');
 
+    if (0 != count($this->helpMessage))
+    {
+      // Convert sorted help message array to string
+      ksort($this->helpMessage);
+      $this->helpMessage = '<p>' . implode('</p><p>', $this->helpMessage) . '</p>';
+    }
```

Comment:
Did initially add them to the relevant field (properties) in the first version, but Analyst team preferred the single pop-up help panel - for most fields. We need to build the help message conditionally depending on view/permissions/settings and I usually stack this type of thing in an array, rather than concatenating a string. No problem to change this to use string if preferred.

Reply:
I think you could render the help messages in a different container, but I see your point. I'd prefer to build the array without keys (you can modify the order of the $names array to get the messages order needed) and the use a foreach in the template appending the paragraph tags, that way there's no need to access sf_data nor unescaping, and I think it's a bit more clear in the code.

Reply:
Have updated in repo. Removed keys and used foreach in template instead of constructing in controller. Had to still use `$sf_data->getRaw` as the array is `sfOutputEscaperArrayDecorator` object in template. Can update this if there's a better way to iterate.

Reply:
Awesome, thanks @PeteVox!